zrouting/routinglookup.go

```diff
@@ -52,13 +52,41 @@ type RoutingLookupTree struct {
 	IPTree  *iptree.IPTree
 }
 
-func (t *RoutingLookupTree) PopulateFromMRT(raw io.Reader) {
+type PathFilter func([]uint32) []uint32
+
+func IdentityPathFilter(path []uint32) []uint32 { return path }
+
+func InternalBGPPathFilter(origin uint32) PathFilter {
+	return func(path []uint32) []uint32 {
+		out := make([]uint32, 0, len(path))
+		var previous uint32
+		previousWasReplaced := false
+		for _, asn := range path {
+			if asn <= 65534 && asn >= 64512 {
+				if previous != origin {
+					out = append(out, origin)
+					previous = origin
+				}
+				previousWasReplaced = true
+			} else {
+				if !previousWasReplaced || asn != previous {
+					out = append(out, asn)
+					previous = asn
+				}
+				previousWasReplaced = false
+			}
+		}
+		return out
+	}
+}
+
+func (t *RoutingLookupTree) PopulateFromMRTFiltered(raw io.Reader, pathFilter PathFilter) {
```

Comment:
I think that you might want to have the ability to replace all the filtered ASNs with the eASN. I think in the Merit data we don't see internal + merit, we only see merit and we need to replace it with the external.